app/services/ipam/base.rb

```diff
@@ -1,23 +1,40 @@
 module IPAM
   class Base
+    extend ActiveModel::Naming
+    extend ActiveModel::Translation
+
     delegate :logger, :to => :Rails
     attr_accessor :mac, :subnet, :excluded_ips
+    attr_reader :errors
 
     def initialize(opts = {})
       @subnet = opts[:subnet]
       @excluded_ips = opts.fetch(:excluded_ips, [])
       @mac = opts.fetch(:mac, nil)
+      @errors = ActiveModel::Errors.new(self)
 
       normalize_mac!
     end
 
+    def suggest_new?
+      true
+    end
+
+    def serializable_hash(options = nil)
```

Comment:
https://github.com/timogoebel/foreman/blob/cdb33fed2b996e6e5fa4d6b610e5ff3a7a449260/app/controllers/concerns/foreman/controller/host_details.rb#L27-L29

`#domain_selected`  in `app/controllers/concerns/foreman/controller/host_details.rb` calls `as_json`, this needs the `serializable_hash` method.